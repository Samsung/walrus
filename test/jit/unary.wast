(module
(func (export "test1") (result i64 i64 i64 i64 i64 i64) (local i64)
  i64.const 0
  local.set 0

  i64.const 0
  i64.clz

  local.get 0
  i64.ctz

  i64.const 0x800000000000
  local.tee 0
  i64.clz

  local.get 0
  i64.ctz

  i64.const 0x100
  local.tee 0
  i64.clz

  local.get 0
  i64.ctz
  (; 64, 64, 16, 47, 55, 8 ;)
)
)

(assert_return (invoke "test1") (i64.const 64) (i64.const 64) (i64.const 16) (i64.const 47) (i64.const 55) (i64.const 8))
