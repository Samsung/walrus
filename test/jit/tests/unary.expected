[[[[[[[  Function   0  ]]]]]]]
Local 0: mapped to: 0 range [1 - 16]
Number of locals reduced from 1 to 1
0: Opcode: i64.const
  Result[0]: i64
1: Opcode: local.set
  Index: 0
  Param[0]: ref:0(0)
2: Opcode: i64.const
  Result[0]: i64
3: Opcode: i64.clz
  Param[0]: ref:2(0)
  Result[0]: i64
4: Opcode: local.get
  Index: 0
  Result[0]: i64
5: Opcode: i64.ctz
  Param[0]: ref:4(0)
  Result[0]: i64
6: Opcode: i64.const
  Result[0]: i64
7: Opcode: local.set
  Index: 0
  Param[0]: ref:6(0)
8: Opcode: local.get
  Index: 0
  Result[0]: i64
9: Opcode: i64.clz
  Param[0]: ref:8(0)
  Result[0]: i64
10: Opcode: local.get
  Index: 0
  Result[0]: i64
11: Opcode: i64.ctz
  Param[0]: ref:10(0)
  Result[0]: i64
12: Opcode: i64.const
  Result[0]: i64
13: Opcode: local.set
  Index: 0
  Param[0]: ref:12(0)
14: Opcode: local.get
  Index: 0
  Result[0]: i64
15: Opcode: i64.clz
  Param[0]: ref:14(0)
  Result[0]: i64
16: Opcode: local.get
  Index: 0
  Result[0]: i64
17: Opcode: i64.ctz
  Param[0]: ref:16(0)
  Result[0]: i64
18: Opcode: return
  Param[0]: ref:3(0)
  Param[1]: ref:5(0)
  Param[2]: ref:9(0)
  Param[3]: ref:11(0)
  Param[4]: ref:15(0)
  Param[5]: ref:17(0)
------------------------------
Frame size: 0, Arguments size: 64 (total: 64)
0: Opcode: i64.const
  Result[0]: i64 unused
1: Opcode: local.set
  Addr: sp[48]
  Param[0]: i64 const:0
2: Opcode: i64.const
  Result[0]: i64 unused
3: Opcode: i64.clz
  Param[0]: i64 const:2
  Result[0]: i64 sp[0]
4: Opcode: local.get
  Addr: sp[48]
  Result[0]: i64 unused
5: Opcode: i64.ctz
  Param[0]: i64 sp[48]
  Result[0]: i64 sp[8]
6: Opcode: i64.const
  Result[0]: i64 unused
7: Opcode: local.set
  Addr: sp[48]
  Param[0]: i64 const:6
8: Opcode: local.get
  Addr: sp[48]
  Result[0]: i64 unused
9: Opcode: i64.clz
  Param[0]: i64 sp[48]
  Result[0]: i64 sp[16]
10: Opcode: local.get
  Addr: sp[48]
  Result[0]: i64 unused
11: Opcode: i64.ctz
  Param[0]: i64 sp[48]
  Result[0]: i64 sp[24]
12: Opcode: i64.const
  Result[0]: i64 unused
13: Opcode: local.set
  Addr: sp[48]
  Param[0]: i64 const:12
14: Opcode: local.get
  Addr: sp[48]
  Result[0]: i64 unused
15: Opcode: i64.clz
  Param[0]: i64 sp[48]
  Result[0]: i64 sp[32]
16: Opcode: local.get
  Addr: sp[48]
  Result[0]: i64 unused
17: Opcode: i64.ctz
  Param[0]: i64 sp[48]
  Result[0]: i64 sp[40]
18: Opcode: return
  Param[0]: i64 sp[0]
  Param[1]: i64 sp[8]
  Param[2]: i64 sp[16]
  Param[3]: i64 sp[24]
  Param[4]: i64 sp[32]
  Param[5]: i64 sp[40]

test1() => i64:64, i64:64, i64:16, i64:47, i64:55, i64:8
